sdk/Pulumi.Tests/Resources/ComponentResourceTests.cs

```diff
@@ -98,10 +100,67 @@ public async void PropagatesProvidersOption()
                 Assert.Equal(provider, custom._provider);
             });
         }
+
+        class BasicComponent : ComponentResource
+        {
+            public Output<string> First { get; set; }
+            [Output]
+            public Output<string> Second { get; set; }
+            [Output("myThird")]
+            public Output<string> Third { get; set; }
+            public BasicComponent(string name) : base("token:token:token", name, ResourceArgs.Empty)
+            {
+                First = Output.Create("first");
+                Second = Output.Create("second");
+                Third = Output.Create("third");
+                RegisterOutputs();
+            }
+        }
+
+        [Fact]
+        public async Task RegisterOutputsCorrectlyUsesReflectionToRegisterOutputProperties()
+        {
+            var mocks = new MinimalMocks();
+            var options = new TestOptions();
+            var resources = await Deployment.TestAsync(mocks, options, () =>
+            {
+                new BasicComponent("basic");
+            });
+
+            foreach (var resource in resources)
+            {
+                if (resource is BasicComponent basic)
+                {
+                    var urn = await basic.Urn.GetValueAsync("<unknown>");
+                    var outputs = mocks.GetRegisteredOutputs(urn) as IDictionary<string, object>;
+                    Assert.Contains("Second", outputs);
+                    Assert.Contains("myThird", outputs);
+
+                    if (outputs["Second"] is Output<string> second)
```

Comment:
Can't we just cast here, if the cast fails it will just fail the test but we expect that. Also catches if we accidentally change this so outputs["second"] becomes null or string.

Reply:
Changed the dictionary to `string -> Output<object?>` in the implementation because the deserializer from `Struct` always gives `OutputData<object?>` which makes casting only needed at the value level after awaiting it.